src/libraries/Common/src/System/Net/Sockets/SocketConnection.cs

```diff
@@ -0,0 +1,101 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.IO.Pipelines;
+using System.Net.Sockets;
+using System.Runtime.ExceptionServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Net.Connections
+{
+    internal sealed class SocketConnection : Connection, IConnectionProperties
+    {
+        private readonly Socket _socket;
+        private readonly SocketsConnectionFactory? _factory;
```

Comment:
How is this going to work when we have SocketListenerFactory as well? Seems like in that case there is no SocketsConnectionFactory/

Reply:
This is only for the case explained in https://github.com/dotnet/runtime/pull/40506#discussion_r468600011. The class is not being prepared to handle `SocketListenerFactory` yet.

Originally used an internal interface here, but  @scalablecory  suggested to avoid interface dispatch in his https://github.com/dotnet/runtime/pull/40506#discussion_r466956225, so ended up with this logic for now.

Probably the dispatch is a cost we are fine to pay to have a cleaner logic here. Want to avoid adding `SocketConnection.Xyz.cs` files just for a few members.